extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridBinaryDataConverter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to BinaryData or BinaryData[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(BinaryData))]
+    [SupportedConverterType(typeof(BinaryData[]))]
+    internal class EventGridBinaryDataConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource == null)
+                {
+                    return new(ConversionResult.Failed(new InvalidOperationException("Context source cannot be null")));
+                }
+                var targetType = context!.TargetType;
```

Comment:
A parameter `null` check would simplify that and provide better diagnostics instead failing with a null reference exception here if this was ever `null` for any reason.

Reply:
I initially had the param check but Lilian had mentioned that context is never going to be null. Is it ok to keep it this way?

Reply:
Being defensive here and having the parameter check helps us identify issues in cases where, even if we don't expect (and particularly those), this is passed in as `null` as we'd get an exception with the detailed information/parameter, making diagnostics simpler. Been hit too many times with null reference exceptions in places where the assumption was that it would never happen.

There are a few scenarios that could potentially invalidate the assumption as the code evolves (including scenarios where the converter may not be directly called by the binding infrastructure).

It's unlikely this would ever be `null`, I'm not against keeping this as-is (principally in this case as it wouldn't be super difficult to identify the issue with a stack), but wanted to share the context above as a defensive approach, even in cases like this, help make the potential issues easier to diagnose.

Reply:
Added check for if context is null!